src/scheduler/Task.php

```diff
@@ -26,8 +26,12 @@
 use pocketmine\utils\Utils;
 
 abstract class Task{
+	/** @phpstan-var TaskHandler<self>|null  */
 	private ?TaskHandler $taskHandler = null;
 
+	/**
+	 * @phpstan-return TaskHandler<self>|null
```

Comment:
This would have to be `static`, not `self`, and I don't think that actually works

Reply:
OK, turns out it actually does work in this instance https://phpstan.org/r/0595cfe5-5ccd-4a13-90da-3b9db2a0e7cb

`static` in generics is just very flaky so I had my doubts